spec/Section 3 -- Type System.md

```diff
@@ -2077,3 +2083,64 @@ to the relevant IETF specification.
 ```graphql example
 scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
 ```
+
+### @defer
+
+```graphql
+directive @defer(label: String, if: Boolean) on FRAGMENT_SPREAD | INLINE_FRAGMENT
+```
+
+The `@defer` directive may be provided for fragment spreads and inline fragments to 
+inform the executor to delay the execution of the current fragment to indicate 
+deprioritization of the current fragment. A query with `@defer` directive will cause
+the request to potentially return multiple responses, where non-deferred data is 
+delivered in the initial response and data deferred is delivered in a subsequent
+response. `@include` and `@skip` take precedence over `@defer`. 
+
+```graphql example
+query myQuery($shouldDefer: Boolean) {
+   user {
+     name
+     ...someFragment @defer(label: 'someLabel', if: $shouldDefer) 
+   }
+}
+fragment someFragment on User {
+  id
+  profile_picture {
+    uri
+  }
+}
+```
+
+#### @defer Arguments
+
+* `if: Boolean` - When `true`, fragment *should* be deferred. When `false`, fragment will not be deferred and data will be included in the initial response. If omitted, defaults to `true`. 
+* `label: String` - A unique label across all `@defer` and `@stream` directives in an operation. This label should be used by GraphQL clients to identify the data from patch responses and associate it with the correct fragments. If provided, the GraphQL Server must add it to the payload.
+
+### @stream
+
+```graphql
+directive @stream(label: String, initialCount: Int = 0, if: Boolean) on FIELD
+```
+
+The `@stream` directive may be provided for a field of `List` type so that the 
+backend can leverage technology such as asynchronous iterators to provide a partial
+list in the initial response, and additional list items in subsequent responses.
+`@include` and `@skip` take precedence over `@stream`.
+
+```graphql example
+query myQuery($shouldStream: Boolean) {
+   user {
+     friends(first: 10) {
+       nodes @stream(label: "friendsStream", initialCount: 5, if: $shouldStream)
+     }
+   }
+}
+```
+
+#### @stream Arguments
+* `if: Boolean` - When `true`, field *should* be streamed. When `false`, the field will not be streamed and all list items will be included in the initial response. If omitted, defaults to `true`. 
+* `label: String` - A unique label across all `@defer` and `@stream` directives in an operation. This label should be used by GraphQL clients to identify the data from patch responses and associate it with the correct fragments. If provided, the GraphQL Server must add it to the payload.
+* `initialCount: Int` - The number of list items the server should return as part of the initial response. If omitted, defaults to `0`. If the value of this argument is less than `0`, it is treated the same as `0`.
```

Comment:
> value of this argument is less than `0`, it is treated the same as `0`.

shouldn't this be a validation error?

I mean it kind of fields heavy to have a validation rule for this but it actually makes no sense to put -10. So wouldn't we be clearer with the user to say that -10 is wrong and the number must be positive or zero?

Reply:
It can't be a validation rule because the value will not be known if it's passed in as a variable. See discussion https://github.com/robrichard/defer-stream-wg/discussions/24

Reply:
This one is resolved with the last spec edit. A field error is thrown when we have a value less than 0.